logging/log_converter.cpp

```diff
@@ -0,0 +1,218 @@
+#include <NewRefereeModule.cpp>
+
+#include <google/protobuf/io/zero_copy_stream_impl.h>
+#include <protobuf/LogFrame.pb.h>
+#include <protobuf/referee.pb.h>
+
+#include <QFile>
+#include <QTextStream>
+
+#include <vector>
+
+#include <fcntl.h>
+#include <time.h>
+
+using namespace Packet;
+using namespace std;
+
+#define BAR "|"
+#define MATCH_ID_LENGTH 32
+
+vector<std::shared_ptr<Packet::LogFrame> > frames;
+
+/**
+ * Defines usage information for launching the Log-Viewer application
+ * @param prog The name of the program
+ */
+void usage(const char* prog) {
+    fprintf(stderr, "Usage: %s <filename.log> <output base filename>\n", prog);
+    exit(1);
+}
+
+/**
+ * Generates random alphanumeric strings using PID and time as the random seed.
+ * A Null terminator is added after the ID
+ * @param id A pointer to a character array that can hold at least length + 1
+ * chars
+ * @param length Length of the ID to be generated.
+ */
+void generateMatchID(char* id, int length) {
+    string chars(
+        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890");
+
+    // Avoid using time based srand init
+    srand(getpid() * time(NULL));
+
+    for (int i = 0; i < length; ++i) {
+        id[i] = chars[rand() % chars.length()];
+    }
+
+    id[length] = '\0';
+}
+
+/**
+ * Read frames and parse protobuf from specified file
+ * @param filename Filename of a RoboJackets protobuf log file
+ * @return true If reading and parsing was flawless
+ * @return false If there was any errors or warnings while parsing
+ */
+bool readFrames(const char* filename) {
+    QFile file(filename);
+
+    if (!file.open(QFile::ReadOnly)) {
+        fprintf(stderr, "Can't open %s: %s\n", filename,
+                (const char*)file.errorString().toLatin1());
+        return false;
+    }
+
+    int n = 0;
```

Comment:
I don't think so. That's legacy from log_viewer. I got rid of it and everything still works. I don't see where it is used at all either. I wonder if at one point it was used to place the frames into an array.